tools/module_selector.py

```diff
@@ -0,0 +1,129 @@
+                )
+            elif version.startswith(">"):
+                selected_modules.extend([f"{module}@{v}" for v in module_versions if Version(v) > Version(version[1:])])
+            elif version.startswith("<"):
+                selected_modules.extend([f"{module}@{v}" for v in module_versions if Version(v) < Version(version[1:])])
+            else:
+                if version in module_versions:
+                    selected_modules.append(f"{module}@{version}")
+                else:
+                    raise ValueError(f"Version {version} of module {module} not found.")
+
+    if random_percentage is not None:
+        percentage = int(random_percentage)
+        if not (0 < percentage <= 100):
+            raise ValueError("Random percentage must be an integer between 1 and 100.")
+
+        total_modules = len(selected_modules)
+        num_to_select = max(1, (percentage * total_modules) // 100)
+        selected_modules = random.sample(selected_modules, num_to_select)
+
+    if not selected_modules:
+        raise ValueError("No matching modules found.")
+
+    return sorted(set(selected_modules))
+
+
+def main():
+    parser = argparse.ArgumentParser(description="Select module versions matching given patterns.")
+    parser.add_argument(
+        "--registry",
+        type=str,
+        default=".",
+        help="Specify the root path of the registry (default: the current working directory).",
+    )
+    parser.add_argument(
+        "--select",
+        action="append",
+        required=True,
+        help=(
+            "Specify module selection patterns in the format <module_pattern>@<version_pattern>. "
+            "The <module_pattern> can include wildcards (*) to match multiple modules. "
+            "The <version_pattern> can be:\n"
+            '  - A specific version (e.g., "1.2.3")\n'
+            '  - "latest" to select the latest version\n'
+            '  - A comparison operator followed by a version (e.g., ">=1.0.0", "<2.0.0")\n'
+            "You can provide multiple --select options to select multiple modules. Examples:\n"
+            '  --select "zlib@latest"\n'
+            '  --select "protobuf@>=27"\n'
+            '  --select "rules_*@<1.5.0"\n'
+            '  --select "*@latest"'
+        ),
+    )
+    parser.add_argument(
+        "--random-percentage",
+        type=int,
+        metavar="PERCENTAGE",
+        help="Percentage of modules to randomly select from the modules matching any of the patterns. Must be an integer between 1 and 100.",
+    )
+
+    args = parser.parse_args()
+
+    registry = RegistryClient(args.registry)
+    module_selections = args.select
+    random_percentage = args.random_percentage
+
+    selected_module_versions = select_modules(registry, module_selections, random_percentage)
+    for module_version in selected_module_versions:
+        print(module_version)
+
+
+if __name__ == "__main__":
+    main()
```

Comment:
it would be nice to chdir to $BAZEL_WORKSPACE_DIRECTORY so this behaves properly under `bazel run`

Reply:
Done, added support for `BUILD_WORKSPACE_DIRECTORY`.